### ISSUE TITLE
Hide warning about the missing hash in metadata.yaml for sample datasets

### ISSUE THREAD

**Issue by sfmig** [neuroinformatics-unit/movement#192]
**Is your feature request related to a problem? Please describe.**
When running `from movement import sample_data` the following warning is thrown from `pooch`.
```bash
Downloading data from 'https://gin.g-node.org/neuroinformatics/movement-test-data/raw/master/metadata.yaml' to file '/Users/nsirmpilatze/.movement/data/temp_metadata.yaml'.
SHA256 hash of downloaded file: 5789de438382fc06310a1fa6cc9fac085a747cd2ad44800ccb019e95ad2f3419
Use this value as the 'known_hash' argument of 'pooch.retrieve' to ensure that the file hasn't changed if it is downloaded again in the future.
```

This is because `pooch` downloads the latest `metadata.yaml` file from GIN and builds a data registry based on that. We don't have a hash associated to that file, because it may change often, and we want to avoid having to update the `movement` code repo, every time a dataset is added or updated on GIN. 

So `pooch` has a vlid reason to complain here, but in `movement`'s context we choose not to care about it.
That said, this warning may look scary to users, especially because it's the first output they see if they run one of our examples.

**Describe the solution you'd like**
Filter out this specific warning so it's not shown.

**Comment by LakshmiSowmya04** [comment]
@sfmig is this issue stil open ? if yes can i work on it ?

**Comment by sfmig** [comment]
Hi @LakshmiSowmya04 thanks for your interest! Yes it is still open and you are welcome to explore it

**Comment by LakshmiSowmya04** [comment]
Okay got it,Thank you

**Comment by LakshmiSowmya04** [comment]
@sfmig I would like to follow this method 
The recommended and permanent solution is to update the pooch.retrieve() call in your sample_data.py file by replacing known_hash=None with the actual SHA256 hash that pooch printed during the download. For example, you can set ``` known_hash="22100193c76c0e3f274e4c986357c95fc645da1f5c67af41a8acfa91313205f4" ```. This will suppress the message entirely and ensure file integrity by verifying the downloaded file each time.

or 

I can add 

```

pooch_logger = logging.getLogger("pooch")
pooch_logger.addFilter(lambda record: "SHA256 hash of downloaded file" not in record.getMessage())

```
so the warning will be ignored when it encounters that line
if this approach is good to go...I am ready to raise a PR.I have tested in my local and seen its not printing anything

Thank you

**Comment by sfmig** [comment]
Hi @LakshmiSowmya04, 
Thanks for looking into this. We do not want to hardcode a specific hash for the `metadata.yaml` file because it often changes (see issue description). The second approach of filtering the warning looks interesting, but we would like to only filter this warning for the specific case of the `metadata.yaml` file, and not any other. 

I am tagging @niksirbi since he has more background for this issue in case he has further thoughts. In any case feel free to open a PR with a draft of the second approach and we can also continue the conversation there. It may be easier to discuss things directly looking at the code.

**Comment by adamltyson** [comment]
Should we just store the hash on GIN and fetch from there? It makes sense to me to check the hash on download. In @brainglobe land, we've had a few issues with interrupted downloads etc

**Comment by niksirbi** [comment]
For us the current setup works (other than the distracting warning). The `metadata.yaml` is a tiny file and unlikely to be interrupted. Even if download fails, the logic in `sample_data.py` is such that we will fall back to a locally cached version of the `metadata.yaml` file and the user will get a [warning](https://github.com/neuroinformatics-unit/movement/blob/f9d728e72892f122e1025bc4476aaedad2ba22f8/movement/sample_data.py#L101) about that. 

So we will only have an issue if the download is interrupted without raising a `RequestException`, which I think is quite unlikely.

> Should we just store the hash on GIN and fetch from there? It makes sense to me to check the hash on download.

That's essentially what the `metadata.yaml` is. It stores the pre-computed hashes for all the data files. If we also pre-compute the hash for `metadata.yaml` itself, we would have to store that somewhere, and the download of that "somewhere" could also fail, etc. *ad infinitum*. 

I think that somehow filtering out this particular warning is simpler, given how unlikely an incident is? I'm still open to other solutions though, I may be failing to see some other way out of this.

**Comment by adamltyson** [comment]
I missed that this was just for metadata.yaml. Agree that filtering is the best option.

Unless we want `meta_metadata.yaml`? 😆

**Comment by LakshmiSowmya04** [comment]
@sfmig @niksirbi @adamltyson thank you for patiently explaining me, i tried to store output and then try to filter the output from sample_data.py file but it turns out to be the line is being printed from ``` local_file_path = pooch.retrieve(
        url=f"{DATA_URL}/{file_name}",
        known_hash=None,
        path=data_dir,
        fname=f"temp_{file_name}",
        progressbar=False,
    ) ``` here **_retrieve_** is being defined in core.py as ``` if known_hash is None:
            get_logger().info(
                "SHA256 hash of downloaded file: %s\n"
                "Use this value as the 'known_hash' argument of 'pooch.retrieve'"
                " to ensure that the file hasn't changed if it is downloaded again"
                " in the future.",
                file_hash(str(full_path)),
            )```
I would like to comment out this part so that output will be like this.

<img width="1350" height="586" alt="Image" src="https://github.com/user-attachments/assets/7b9e4226-f7dd-48ab-a162-2f231ab69500" />

**Comment by niksirbi** [comment]
Hey @LakshmiSowmya04, you are right about the source of this warning message, but we can't simply solve it by commenting out that line.

The `core.py` file you are referring to is in the source code of `pooch` itself and we don't want to modify that.
Any changes we make should be solely confined to `movement` itself, and in this case the `sample_data.py` file.

We have to somehow find a way to specifically silence that INFO message during that particular `pooch.retrieve` call.

**Comment by LakshmiSowmya04** [comment]
> Hey @LakshmiSowmya04, you are right about the source of this warning message, but we can't simply solve it by commenting out that line.
> 
> The `core.py` file you are referring to is in the source code of `pooch` itself and we don't want to modify that.
> Any changes we make should be solely confined to `movement` itself, and in this case the `sample_data.py` file.
> 
> We have to somehow find a way to specifically silence that INFO message during that particular `pooch.retrieve` call.

If I'm not wrong we need to hide this warning for few files form movement right ? So is it fine to add a condition in core.py itself and then hide ?

**Comment by niksirbi** [comment]
> If I'm not wrong we need to hide this warning for few files form movement right ? So is it fine to add a condition in core.py itself and then hide ?

No, that wouldn’t be the right place to make the change. The file `core.py` belongs to **pooch**, which is an *external* package that `movement` uses to download data from the internet. We should never modify code that comes from another package — it’s like editing someone else’s library instead of just using it.

What we want is to adjust how **movement** uses pooch, not pooch itself.
And we only need to do that in one place — in the file `sample_data.py`.

All data downloads in movement eventually go through that file, and specifically through [this line](https://github.com/neuroinformatics-unit/movement/blob/7a1499df2222d1c45ec5e6401c1b436106e49e06/movement/sample_data.py#L57), which triggers the info message you’ve seen.

So if we handle the warning there (for example, by adding a small condition to silence it only for the `metadata.yaml` file), the fix will automatically apply everywhere that `movement` uses sample data.

If you’re new to how importing works in Python, this [tutorial](https://realpython.com/python-modules-packages/) may help to clarify concepts around modules and packages, which can be confusing.

**Comment by LakshmiSowmya04** [comment]
> > If I'm not wrong we need to hide this warning for few files form movement right ? So is it fine to add a condition in core.py itself and then hide ?
> 
> No, that wouldn’t be the right place to make the change. The file `core.py` belongs to **pooch**, which is an *external* package that `movement` uses to download data from the internet. We should never modify code that comes from another package — it’s like editing someone else’s library instead of just using it.
> 
> What we want is to adjust how **movement** uses pooch, not pooch itself.
> And we only need to do that in one place — in the file `sample_data.py`.
> 
> All data downloads in movement eventually go through that file, and specifically through [this line](https://github.com/neuroinformatics-unit/movement/blob/7a1499df2222d1c45ec5e6401c1b436106e49e06/movement/sample_data.py#L57), which triggers the info message you’ve seen.
> 
> So if we handle the warning there (for example, by adding a small condition to silence it only for the `metadata.yaml` file), the fix will automatically apply everywhere that `movement` uses sample data.
> 
> If you’re new to how importing works in Python, this [tutorial](https://realpython.com/python-modules-packages/) may help to clarify concepts around modules and packages, which can be confusing. 

Ohh...I thought we can modify for our issue....thank you,now I got clarity :)

**Comment by Debajeet-1411** [comment]
Hi! I’d like to work on this issue. 
Based on the discussion above, I will implement the temporary logger filter inside sample_data.py 
so that the SHA256 message is suppressed only for “metadata.yaml” and only during that retrieve call.

Please let me know if this is okay. I will begin working on it.

**Comment by niksirbi** [comment]
This issue has been fixed by #695.
